sky/skylet/providers/lambda_cloud/node_provider.py

```diff
@@ -51,80 +53,84 @@ def __init__(self, provider_config: Dict[str, Any],
                 self.metadata[node['id']] = {
                     'tags': {
-                        TAG_RAY_CLUSTER_NAME: cluster_name,
+                        TAG_RAY_CLUSTER_NAME: self.cluster_name,
                         TAG_RAY_NODE_STATUS: STATUS_UP_TO_DATE,
                         TAG_RAY_NODE_KIND: NODE_KIND_HEAD,
                         TAG_RAY_USER_NODE_TYPE: 'ray_head_default',
-                        TAG_RAY_NODE_NAME: f'ray-{cluster_name}-head',
-                        TAG_RAY_LAUNCH_CONFIG: launch_hash,
+                        TAG_RAY_NODE_NAME: f'ray-{self.cluster_name}-head',
+                    }
+                }
+            elif node['name'] == f'{self.cluster_name}-worker':
+                self.metadata[node['id']] = {
+                    'tags': {
+                        TAG_RAY_CLUSTER_NAME: self.cluster_name,
+                        TAG_RAY_NODE_STATUS: STATUS_UP_TO_DATE,
+                        TAG_RAY_NODE_KIND: NODE_KIND_WORKER,
+                        TAG_RAY_USER_NODE_TYPE: 'ray_worker_default',
+                        TAG_RAY_NODE_NAME: f'ray-{self.cluster_name}-worker',
                     }
                 }
 
     def _list_instances_in_cluster(self) -> Dict[str, Any]:
         """List running instances in cluster."""
         vms = self.lambda_client.list_instances()
-        return [node for node in vms if node.get('name') == self.cluster_name]
+        possible_names = [
+            f'{self.cluster_name}-head', f'{self.cluster_name}-worker'
+        ]
+        return [node for node in vms if node.get('name') in possible_names]
 
     @synchronized
     def _get_filtered_nodes(self, tag_filters: Dict[str,
                                                     str]) -> Dict[str, Any]:
 
-        def match_tags(vm):
+        def _extract_metadata(vm: Dict[str, Any]) -> Dict[str, Any]:
+            metadata = {'id': vm['id'], 'status': vm['status'], 'tags': {}}
+            instance_info = self.metadata[vm['id']]
+            if instance_info is not None:
+                metadata['tags'] = instance_info['tags']
+            ip = vm['ip']
+            metadata['external_ip'] = ip
+            return metadata
+
+        def _match_tags(vm: Dict[str, Any]):
             vm_info = self.metadata[vm['id']]
             tags = {} if vm_info is None else vm_info['tags']
             for k, v in tag_filters.items():
                 if tags.get(k) != v:
                     return False
             return True
 
+        def _get_internal_ip(node: Dict[str, Any]):
+            runner = command_runner.SSHCommandRunner(node['external_ip'],
+                                                     'ubuntu',
+                                                     self.ssh_key_path)
+            out = runner.run(GET_INTERNAL_IP_CMD,
+                             require_outputs=True,
+                             stream_logs=False)
+            assert out[0] == 0
```

Comment:
Do we want to raise an error and have more information in the error message to make it easier to debug in the future? We can use `subprocess_utils.handle_returncode()`

Reply:
Good idea.

Reply:
Done.